clarity/src/vm/types/mod.rs

```diff
@@ -17,472 +17,54 @@
-        }
-
-        pub fn destruct(self) -> (u8, [u8; 20]) {
-            let Self(version, bytes) = self;
-            (version, bytes)
-        }
-    }
-
-    impl fmt::Display for StandardPrincipalData {
-        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-            let c32_str = self.to_address();
-            write!(f, "{c32_str}")
-        }
-    }
-
-    impl fmt::Debug for StandardPrincipalData {
-        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-            let c32_str = self.to_address();
-            write!(f, "StandardPrincipalData({c32_str})")
-        }
-    }
-}
-
-#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize, PartialOrd, Ord)]
-pub struct QualifiedContractIdentifier {
-    pub issuer: StandardPrincipalData,
-    pub name: ContractName,
-}
-
-impl QualifiedContractIdentifier {
-    pub fn new(issuer: StandardPrincipalData, name: ContractName) -> QualifiedContractIdentifier {
-        Self { issuer, name }
-    }
-
-    pub fn local(name: &str) -> Result<QualifiedContractIdentifier> {
-        let name = name.to_string().try_into()?;
-        Ok(Self::new(StandardPrincipalData::transient(), name))
-    }
-
-    #[allow(clippy::unwrap_used)]
-    pub fn transient() -> QualifiedContractIdentifier {
-        let name = String::from("__transient").try_into().unwrap();
-        Self {
-            issuer: StandardPrincipalData::transient(),
-            name,
-        }
-    }
-
-    /// Was this contract issued by the null issuer address? (i.e., is it a "boot contract")
-    pub fn is_boot(&self) -> bool {
-        self.issuer.1 == [0; 20]
-    }
-
-    pub fn parse(literal: &str) -> Result<QualifiedContractIdentifier> {
-        let split: Vec<_> = literal.splitn(2, '.').collect();
-        if split.len() != 2 {
-            return Err(RuntimeErrorType::ParseError(
-                "Invalid principal literal: expected a `.` in a qualified contract name"
-                    .to_string(),
-            )
-            .into());
-        }
-        let sender = PrincipalData::parse_standard_principal(split[0])?;
-        let name = split[1].to_string().try_into()?;
-        Ok(QualifiedContractIdentifier::new(sender, name))
-    }
-}
-
-impl fmt::Display for QualifiedContractIdentifier {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}.{}", self.issuer, self.name)
-    }
-}
-
-#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
-pub enum PrincipalData {
-    Standard(StandardPrincipalData),
-    Contract(QualifiedContractIdentifier),
-}
-
-pub enum ContractIdentifier {
-    Relative(ContractName),
-    Qualified(QualifiedContractIdentifier),
+    pub use clarity_serialization::types::StandardPrincipalData;
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub struct OptionalData {
-    pub data: Option<Box<Value>>,
+pub trait GetAtomValues {
```

Comment:
Is there a reason this trait needs to exist?  For example, is there any code that takes a type parameter `T: GetAtomValues`?  If there's no reason other than the fact that some types implement a function `atom_values()`, then please remove this trait and revert its `impl`s.

Reply:
Ah, I see that `SymbolicExpression` does not exist in `clarity-serialization`.  It may be difficult to add it because there is code in `SymbolicExpression` and its dependencies that is gated on Clarity-specific features (e.g. `developer-mode`).

If we're keeping `SymbolicExpression` in the `clarity` crate (which is probably better than trying to move it), then please instead name this trait something indicative of the need, like `SymbolicExpressionExtensions`.  Also, please add a docstring indicating that this trait exists in order to extend enums and structs from `clarity-serialization` to interface with `SymbolicExpression` and its ilk.

Reply:
Actually, I no longer think it's a good idea to keep `SymbolicExpression` in `clarity`, now that I see how the traits have worked out.  See https://github.com/stacks-network/stacks-core/pull/6310/files#r2252452968